picard/mbjson.py

```diff
@@ -101,8 +101,15 @@ def _decamelcase(text):
 _BLANK_SPECIAL_RELTYPES = {'vocal': 'vocals'}
 
 
-def _parse_attributes(attrs, reltype):
-    attrs = [_decamelcase(_REPLACE_MAP.get(a, a)) for a in attrs]
+def _transform_attribute(attr, credits):
+    if attr in credits:
+        return credits[attr]
+    else:
+        return _decamelcase(_REPLACE_MAP.get(attr, attr))
+
+
+def _parse_attributes(attrs, reltype, credits):
+    attrs = [_transform_attribute(a, credits) for a in attrs]
```

Comment:
There are 3 list comprehensions  based on `attrs` here, i wonder if this can be rewritten to something simpler using a normal loop.